components/browser/search/src/test/java/mozilla/components/browser/search/provider/AssetsSearchEngineProviderTest.kt

```diff
@@ -0,0 +1,120 @@
+
+@RunWith(RobolectricTestRunner::class)
+class AssetsSearchEngineProviderTest {
+    @Test
+    fun `Load search engines for en-US from assets`() = runBlocking {
+        val localizationProvider = object : SearchLocalizationProvider() {
+            override val country: String = "US"
+            override val language = "en"
+            override val region: String? = null
+        }
+
+        val searchEngineProvider = AssetsSearchEngineProvider(localizationProvider)
+
+        val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+        assertEquals(6, searchEngines.size)
+    }
+
+    @Test
+    fun `Load search engines for en-US with with filter`() = runBlocking {
+        val localizationProvider = object : SearchLocalizationProvider() {
+            override val country: String = "US"
+            override val language = "en"
+            override val region: String? = null
+        }
+
+        val filter = object : SearchEngineFilter {
+            private val exclude = listOf("yahoo", "bing", "ddg")
+
+            override fun filter(searchEngine: SearchEngine): Boolean {
+                return !exclude.contains(searchEngine.identifier)
+            }
+        }
+
+        val searchEngineProvider = AssetsSearchEngineProvider(localizationProvider, listOf(filter))
+
+        val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+        assertEquals(4, searchEngines.size)
+    }
+
+    @Test
+    fun `Load search engines for de-DE with US region override`() = runBlocking {
+        run {
+            val localizationProviderWithoutRegion = object : SearchLocalizationProvider() {
+                override val country: String = "DE"
+                override val language = "de"
+                override val region: String? = null
+            }
+
+            val searchEngineProvider = AssetsSearchEngineProvider(localizationProviderWithoutRegion)
+            val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+            assertEquals(7, searchEngines.size)
+            assertContainsSearchEngine("google", searchEngines)
+            assertContainsNotSearchEngine("google-2018", searchEngines)
+        }
+
+        run {
+            val localizationProviderWithRegion = object : SearchLocalizationProvider() {
+                override val country: String = "DE"
+                override val language = "de"
+                override val region: String? = "US"
+            }
+
+            val searchEngineProvider = AssetsSearchEngineProvider(localizationProviderWithRegion)
+            val searchEngines = searchEngineProvider.loadSearchEngines(RuntimeEnvironment.application)
+
+            assertEquals(7, searchEngines.size)
+            assertContainsSearchEngine("google-2018", searchEngines)
+            assertContainsNotSearchEngine("google", searchEngines)
+        }
+    }
+
+    @Test
```

Comment:
\[detekt\] INFO: \[style/NamingConventionViolation\] All names in the codebase should be matching the naming convention of the codebase\.